packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -214,9 +214,7 @@ class Scaffold extends StatefulWidget {
     this.drawer,
     this.scrollableKey,
     this.appBarBehavior: AppBarBehavior.anchor
-  }) : super(key: key) {
-    assert((appBarBehavior == AppBarBehavior.scroll) ? scrollableKey != null : true);
-  }
+  }) : super(key: key);
 
```

Comment:
Yes, I can add that. Hopefully we'll be able to obviate scollableKey altogether, but for now.